keras_cv/models/object_detection_3d/center_pillar.py

```diff
@@ -56,31 +51,31 @@ def __init__(
         prediction_decoder,
         **kwargs,
     ):
-        point_xyz = keras.layers.Input((None, 3), name="point_xyz")
-        point_feature = keras.layers.Input((None, 4), name="point_feature")
-        point_mask = keras.layers.Input(
-            (None, 1), name="point_mask", dtype=tf.bool
-        )
-
-        inputs = {
-            "point_xyz": point_xyz,
-            "point_feature": point_feature,
-            "point_mask": point_mask,
-        }
-
-        voxel_feature = voxel_net(
-            point_xyz, point_feature, tf.squeeze(point_mask, axis=-1)
-        )
-        voxel_feature = backbone(voxel_feature)
-        predictions = multiclass_head(voxel_feature)
-
-        super().__init__(inputs=inputs, outputs=predictions, **kwargs)
-
+        super().__init__(**kwargs)
         self._voxelization_layer = voxel_net
-        self._backbone = backbone
+        self._unet_layer = backbone
         self._multiclass_head = multiclass_head
         self._prediction_decoder = prediction_decoder
         self._head_names = self._multiclass_head._head_names
+        self.built = True
+
+    def call(self, input_dict, training=None):
```

Comment:
This should also still be a functional subclass model and we should drop the `call`